common/scripted_effects/iisc_science_ship_effects.txt

```diff
@@ -28,123 +28,134 @@ iisc_create_science_ship_fleet = {
+				OR = {
+					is_fleet_idle = yes
 					NOT = {
-						has_star_flag = iisc_being_surveyed_for_@prev.owner.overlord
-					}
-					any_system_planet = {
-						is_surveyed = {
-							who = prevprev.owner
-							status = no
-						}
+						has_fleet_flag = iisc_science_ship_has_dest
 					}
 				}
 			}
-			iisc_science_ship_find_local_survey_target = yes
-		}
-		else_if = {
-			limit = {
-				event_target:iisc_hq_planet = {
-					any_system = {
-						has_access_fleet = prevprev.owner
+			if = {
+				limit = {
+					solar_system = {
 						NOT = {
-							has_star_flag = iisc_being_surveyed_for_@prevprev.owner.overlord
-						}
-						distance = {
-							source = prev
-							max_jumps = @iisc_science_ship_close_system
-							type = hyperlane
+							has_star_flag = iisc_being_surveyed_for_@prev.owner.overlord
 						}
 						any_system_planet = {
 							is_surveyed = {
-								who = prevprevprev.owner
+								who = prevprev.owner
 								status = no
 							}
 						}
 					}
 				}
+				iisc_science_ship_find_local_survey_target = yes
+			}
+			else_if = {
+				limit = {
+					event_target:iisc_hq_planet = {
+						any_system = {
+							has_access_fleet = prevprev.owner
+							NOT = {
+								has_star_flag = iisc_being_surveyed_for_@prevprev.owner.overlord
+							}
+							distance = {
+								source = prev
+								max_jumps = @iisc_science_ship_close_system
+								type = hyperlane
+							}
+							any_system_planet = {
+								is_surveyed = {
+									who = prevprevprev.owner
+									status = no
+								}
+							}
+						}
+					}
+				}
+				iisc_science_ship_find_closest_survey_target = yes
 			}
-			iisc_science_ship_find_closest_survey_target = yes
-		}
-	}
-	if = {
-		limit = {
-			is_fleet_idle = yes
-			has_fleet_flag = iisc_science_ship_has_dest
-			exists = event_target:iisc_science_ship_dest
 		}
-		clear_fleet_actions = this
-		queue_actions = {
-			move_to = event_target:iisc_science_ship_dest
-			orbit_planet = event_target:iisc_science_ship_dest
+		if = {
+			limit = {
+				is_fleet_idle = yes
+				has_fleet_flag = iisc_science_ship_has_dest
+				exists = event_target:iisc_science_ship_dest
+			}
+			clear_fleet_actions = this
+			queue_actions = {
+				move_to = event_target:iisc_science_ship_dest
+				orbit_planet = event_target:iisc_science_ship_dest
+			}
 		}
-	}
-	if = {
-		limit = {
-			owner = {
-				iisc_should_destroy_science_ship = yes
+		if = {
+			limit = {
+				owner = {
+					iisc_should_destroy_science_ship = yes
+				}
 			}
+			delete_fleet = this
 		}
-		delete_fleet = this
-	}
-	else = {
-		save_event_target_as = iisc_science_ship_fleet
-		owner = {
-			country_event = {
-				id = iisc_science_ships.10
-				days = 30
-				random = 30
+		else = {
+			save_event_target_as = iisc_science_ship_fleet
+			owner = {
+				country_event = {
+					id = iisc_science_ships.10
+					days = 30
+					random = 30
+				}
```

Comment:
This looks like a mess, but all I did was add an if statement surrounded everything in the trigger requiring there to be the right flags present :|